src/Formik.tsx

```diff
@@ -476,36 +469,55 @@ export function useFormik<Values = object>({
     [fields]
   );
 
-  const handleBlur = React.useCallback((eventOrString: any):
-    | void
-    | ((e: any) => void) => {
-    if (isString(eventOrString)) {
-      return event => executeBlur(event, eventOrString);
-    } else {
-      executeBlur(eventOrString);
-    }
+  const setTouched = React.useCallback(
+    (touched: FormikTouched<Values>) => {
+      dispatch({ type: 'SET_TOUCHED', payload: touched });
+      return validateOnBlur
```

Comment:
This is how it works now in v1. tbh, `validateOnBlur` really means `validateOnTouched` the more i think about it.

Reply:
Well, as I said, it would be nice to give the option to opt-out of this behavior for a `setTouched` call. In case I really do want to set touched for one field and avoid validation for some reason. I don't have a real use case though, so perhaps it's not worth it.

Reply:
I agree. If we didn't need to keep v2 backwards compat, i would not have this side-effect stuff. This is something we will fix for v3.